### PR TITLE
Learn difference between Char and Wchar_t

### PR DESCRIPTION
**Pull request by TimurKM**
Please, review

### CODE REVIEW

File: TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	str = "Hello, new world!"; // changes work correctly
+	std::cout << str << std::endl;
+	std::cout << sizeof(str) << std::endl; // sizeof work correctly
+	// std::cout << _countof(str) << std::endl; // it doesn't work, I haven't found how to connect it through the library
+
+	const wchar_t* str1 = L"Hello, world!";
+	std::wstring string(str1); // it does not work correctly, numbers are displayed instead of the phrase
+	std::cout << str1 << std::endl;
+	wchar_t* p1;
+	p1 = const_cast<wchar_t*>(str1);
```
Comment (dimagerasimov) [inline TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp:20]: the same as for the "p".

File: TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#include <iostream>
+#include <stdlib.h>
```
Comment (dimagerasimov) [inline TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp:2]: what do we include from stdlib?

File: TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#include <string>
+
+int main()
+{
+	const char* str = "Hello, world!"; // it works correctly
```
Comment (dimagerasimov) [inline TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp:7]: please declare the string like const char str[] =. The same for wchar_t down the code..

**Review by dimagerasimov — CHANGES_REQUESTED** [review]

File: TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	std::cout << str << std::endl;
+	char* p;
+	p = const_cast<char*>(str);
```
[the remark this answers is not in the document]
Comment (dimagerasimov) [reply TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp:10]: The example became more complicated than I expected in the beginning. Not to confuse people, please remove the variables 'p' and 'p1'. The main takeaway here, we now know that the literals "Hello, world!" may be stored in read-only segments of memory, so we never touch it.

[the remark this answers is not in the document]
Comment (TimurKM) [reply TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp]: These segments were deleted

File: TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
+#include <iostream>
+#define COUNTOF(a) (sizeof(a) / sizeof(a[0]))
+
+int main()
+{
+	const char str[] = "Hello, world!";
+	const char* same_str = "Hello, world!";
+	std::cout << "Our str: " << str << std::endl;
+	std::cout << "Sizeof: " << sizeof(str) << std::endl;
```
Comment (dimagerasimov) [inline TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp:9]: Forgot sizeof(same_str) and sizeof(same_str1). Please add. Explain why results are different respectively from sizeof(str), sizeof(str1).

Comment (TimurKM) [reply TaskWithCharAndWchar_t/TaskWithCharAndWchar_t.cpp]: Added

**Review by dimagerasimov — APPROVED** [review]

**Pull request merged**